InsightsBackend/src/main/java/org/frankframework/insights/label/LabelService.java

```diff
@@ -106,16 +119,39 @@ private List<Label> getLabelsByReleaseId(String releaseId) throws ReleaseNotFoun
                 .collect(Collectors.toList());
     }
 
-    /**
-     * Filters labels based on priority defined in the GitHub properties.
-     * @param labels the list of labels to filter
-     * @return a set of labels that match the priority criteria
-     */
-    private Set<Label> filterLabelsByPriority(List<Label> labels) {
-        return labels.stream()
-                .filter(label -> priorityLabels.contains(label.getColor()))
-                .collect(Collectors.toSet());
-    }
+	private Map<String, Long> countLabelOccurrences(List<Label> labels) {
+		return labels.stream()
+				.collect(Collectors.groupingBy(Label::getId, Collectors.counting()));
+	}
+
+	private Map<String, Label> mapUniqueLabels(List<Label> labels) {
+		return labels.stream()
+				.collect(Collectors.toMap(Label::getId, Function.identity(), (l1, _) -> l1));
+	}
+
+	private List<Label> getSortedLabelsByPriorityAndCount(
+			Map<String, Label> uniqueLabelMap,
+			Map<String, Long> labelCountMap) {
+		return uniqueLabelMap.values().stream()
+				.filter(label -> priorityLabels.contains(label.getColor()))
+				.filter(label -> !ignoredLabels.contains(label.getColor()))
```

Comment:
Cant this be merged into one `.filter()`?
```suggestion
				.filter(label -> priorityLabels.contains(label.getColor()) && !ignoredLabels.contains(label.getColor()))
```